PIVX-Labs/MyPIVXWallet
# switch to websockets

## Abstract

Replace the legacy HTTP requests based system with a websocket implementation.

---

## Testing

At the moment this is testable only on mainnet (Duddino's testnet explorer needs to be updated still).
What I suggest for testing:

-  Activate debug mode, so you can see more logs from the console
- Try to sync your mainnet wallet from zero. How fast it is? Is it faster than the current HTTP version?
- Test transaction creation, promo codes, etc... (basically everything involving network requests)
- Try to  change explorer from duddino to zkbitcoin or viceversa. verify that nothing gets broken.
- Finally keep the wallet open for a long period of time (let's say 1 hour). From console see how many times the connection got closed and verify that network is still working
